app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,27 @@
+# controllers/users/registrations_controller.rb
+class Users::RegistrationsController < Devise::RegistrationsController
+
+  before_filter :configure_permitted_parameters
+
+  #POST
+  def create
+    #byebug #debugging
+    super
+  @user.add_role(params[:role_name])
```

Comment:
if condition that no validates error... persisted... opposite of testnewrecord,, id